### PR TITLE
Joshua/fixing inconsistent naming

### LINKED ISSUE #1800: Fix naming inconsistent with code style guides
### Description of the task
There are areas of inconsistent naming. Here are all known issues:
* There's inconsistency in how this is uses in snake_case contexts. Since these are two separate words, there should be an "_" between the words: 
  * replace gamestate with game_state
  * replace freekick with free_kick
* rename `deflect_off_enemy_target` to `camelCase`

### Acceptance criteria

- [x] replace gamestate with game_state
- [x] replace freekick with free_kick
- [x] rename `deflect_off_enemy_target` to `camelCase`

### PR DESCRIPTION
**Pull request by JoshuaWu7**
### Description

Refactors the inconsistent naming based on code style. Refer to ticket on refactored code.

### Resolved Issues

resolves #1800 

### Review Checklist

**_It is the reviewers responsibility to also make sure every item here has been covered_**

- [x] **Function & Class comments**: All function definitions (usually in the `.h` file) should have a javadoc style comment at the start of them. For examples, see the functions defined in `thunderbots/software/geom`. Similarly, all classes should have an associated Javadoc comment explaining the purpose of the class.
- [ ] **Remove all commented out code**
- [x] **Remove extra print statements**: for example, those just used for testing
- [x] **Resolve all TODO's**: All `TODO` (or similar) statements should either be completed or associated with a github issue
- [ ] **Justify drops in code coverage**: If this PR results in a non-trivial drop in code coverage (a bot should post a coverage diagram as a comment), please justify why we can't test the code that's not covered.

### CODE REVIEW

File: docs/software-architecture-and-design.md [hunk 1]
```diff
@@ -10,7 +10,7 @@
     * [Robot](#robot)
     * [Ball](#ball)
     * [Field](#field)
-    * [GameState](#gamestate)
+    * [GameState](#game_state)
```
Comment (jonathanlew) [inline docs/software-architecture-and-design.md]: please check that the link still works

File: src/software/ai/evaluation/BUILD [hunk 2]
```diff
@@ -54,9 +54,9 @@ cc_test(
 )
 
 cc_library(
-    name = "deflect_off_enemy_target",
-    srcs = ["deflect_off_enemy_target.cpp"],
-    hdrs = ["deflect_off_enemy_target.h"],
+    name = "deflectOffEnemyTarget",
+    srcs = ["deflectOffEnemyTarget.cpp"],
+    hdrs = ["deflectOffEnemyTarget.h"],
```
Comment (jonathanlew) [inline src/software/ai/evaluation/BUILD]: file names follow snake_case, so please don't change this

File: src/software/ai/evaluation/BUILD [hunk 3]
```diff
@@ -67,10 +67,10 @@ cc_library(
 )
 
 cc_test(
-    name = "deflect_off_enemy_target_test",
-    srcs = ["deflect_off_enemy_target_test.cpp"],
+    name = "deflectOffEnemyTarget_test",
+    srcs = ["deflectOffEnemyTarget_test.cpp"],
     deps = [
-        ":deflect_off_enemy_target",
+        ":deflectOffEnemyTarget",
```
Comment (jonathanlew) [inline src/software/ai/evaluation/BUILD]: same here

File: src/software/ai/hl/stp/tactic/shadow_freekicker_tactic.h [hunk 4]
```diff
@@ -14,7 +14,7 @@ class ShadowFreekickerTactic : public Tactic
      * This enum indicates the side the robot running this tactic should shadow on. "Left"
      * and "Right" are from the POV of a robot in the friendly goal looking at the enemy
      * taking the free kick. For example, the tactic using the LEFT enum would shadow
-     * slightly to the left of the vector from the enemy freekicker to the friendly goal
+     * slightly to the left of the vector from the enemy free_kicker to the friendly goal
```
Comment (jonathanlew) [inline src/software/ai/hl/stp/tactic/shadow_freekicker_tactic.h]: where there are comments like here, we should use a space
```suggestion
     * slightly to the left of the vector from the enemy free kicker to the friendly goal
```

File: src/software/gui/full_system/widgets/ai_control.cpp [hunk 5]
```diff
@@ -121,6 +121,19 @@ void setupDefendingSideComboBox(
 }
 
 void setupGameStateOverrideComboBox(
+<<<<<<< HEAD
```
Comment (jonathanlew) [inline src/software/gui/full_system/widgets/ai_control.cpp]: these are merge conflict markers and these need to be cleaned up

**Review by jonathanlew — CHANGES_REQUESTED** [review]
Going in the right direction, but there are a few issues

File: .vscode/settings.json [hunk 6]
```diff
@@ -0,0 +1,3 @@
+{
+    "bazel.enableCodeLens": true
+}
```
Comment (kelvinkoon) [inline .vscode/settings.json]: I don't know if this should be here?

**Review by kelvinkoon — CHANGES_REQUESTED** [review]

File: docs/software-architecture-and-design.md (hunk 1 above)
Comment (JoshuaWu7) [reply docs/software-architecture-and-design.md]: Link is still working

File: src/software/simulated_tests/BUILD [hunk 7]
```diff
@@ -123,8 +123,12 @@ cc_test(
     srcs = ["enemy_free_kick_play_test.cpp"],
     deps = [
         ":simulated_test_fixture",
+<<<<<<< HEAD
```
Comment (JoshuaWu7) [inline src/software/simulated_tests/BUILD]: Merge conflict markers

File: src/software/gui/full_system/widgets/ai_control.h [hunk 8]
```diff
@@ -56,6 +56,17 @@ void setupDefendingSideComboBox(
 /**
  * Sets up the ComboBox widget that will be used to override the AI GameState
  *
+<<<<<<< HEAD
+ * @param widget The widget containing the ComboBox to set up
```
Comment (JoshuaWu7) [inline src/software/gui/full_system/widgets/ai_control.h]: Merge conflict

File: src/software/gui/full_system/widgets/ai_control.cpp [hunk 9]
```diff
@@ -130,29 +143,34 @@ void setupGameStateOverrideComboBox(
...
         else
         {
-            gamestate_override_parameter->setValue(true);
+            game_state_override_parameter->setValue(true);
             previous_game_state_parameter->setValue(
                 current_game_state_parameter->value());
             current_game_state_parameter->setValue(text.toStdString());
         }
     };
+<<<<<<< HEAD
```
Comment (JoshuaWu7) [inline src/software/gui/full_system/widgets/ai_control.cpp]: Merge conflict

**Review by jordanzliu — COMMENTED** [review]
LGTM other than merge conflict markers

File: src/software/gui/full_system/widgets/ai_control.h [hunk 10]
```diff
@@ -56,15 +56,15 @@ void setupDefendingSideComboBox(
...
- * @param gamestate_override_combo_box The ComboBox to set up
- * @param gamestate_override_parameter The parameter that controls whether or not
- * we override the gamestate
- * @param previous_game_state_parameter The parameter that holds the previous gamestate
- * @param current_game_state_parameter The parameter that holds the current gamestate
+ * @param game_state_override_combo_box The ComboBox to set up
+ * @param game_state_override_parameter The parameter that controls whether or not
+ * we override the game_state
+ * @param previous_game_state_parameter The parameter that holds the previous game_state
+ * @param current_game_state_parameter The parameter that holds the current game_state
```
Comment (jonathanlew) [inline src/software/gui/full_system/widgets/ai_control.h]:
```suggestion
 * @param previous_game_state_parameter The parameter that holds the previous game state
 * @param current_game_state_parameter The parameter that holds the current game state
```

File: src/software/ai/motion_constraint/motion_constraint_set_builder_test.cpp [hunk 11]
```diff
@@ -77,7 +77,7 @@ namespace
             std::pair<std::shared_ptr<Tactic>, std::set<MotionConstraint>>(
                 new StopTactic(false), std::set<MotionConstraint>({}))};
 
-    // sets of motion constraints for each type of gamestate
+    // sets of motion constraints for each type of game_state
```
Comment (jonathanlew) [inline src/software/ai/motion_constraint/motion_constraint_set_builder_test.cpp]:
```suggestion
    // sets of motion constraints for each type of game state
```

File: src/software/ai/motion_constraint/motion_constraint_set_builder.h [hunk 12]
```diff
@@ -23,7 +23,7 @@ std::set<MotionConstraint> buildMotionConstraintSet(const GameState &game_state,
 
 /**
  * Builds a set of motion constraints by adding motion constraints determined from
- * gamestate
+ * game_state
```
Comment (jonathanlew) [inline src/software/ai/motion_constraint/motion_constraint_set_builder.h]:
```suggestion
 * game state
```

File: src/software/ai/hl/stp/tactic/shadow_free_kicker_tactic_test.cpp [hunk 13]
```diff
@@ -18,7 +17,7 @@ class ShadowFreekickerTacticTest : public testing::Test
      * @param enemy_robot_positions: a vector of Points to represent the position of
      * robots
      * @param left_or_right: a enum argument in the shadowFreekickerTactic to represent
-     * whether the robot is  blocking the freekick from left or right
+     * whether the robot is  blocking the free_kick from left or right
```
Comment (jonathanlew) [inline src/software/ai/hl/stp/tactic/shadow_free_kicker_tactic_test.cpp]:
```suggestion
     * whether the robot is  blocking the free kick from left or right
```

File: src/software/ai/hl/stp/tactic/shadow_free_kicker_tactic_test.cpp [hunk 14]
```diff
@@ -73,7 +72,7 @@ class ShadowFreekickerTacticTest : public testing::Test
      * distances needed for unit tests
      * @param ball_position: the position of the ball in Point format
      * @param left_or_right: a enum argument in the shadowFreekickerTactic to represent
-     * whether the robot is  blocking the freekick from left or right
+     * whether the robot is  blocking the free_kick from left or right
```
Comment (jonathanlew) [inline src/software/ai/hl/stp/tactic/shadow_free_kicker_tactic_test.cpp]:
```suggestion
     * whether the robot is  blocking the free kick from left or right
```

File: src/software/ai/hl/stp/tactic/shadow_free_kicker_tactic.h [hunk 15]
```diff
@@ -65,6 +65,6 @@ class ShadowFreekickerTactic : public Tactic
     Field field;
 
     const double FREE_KICK_MAX_PROXIMITY =
-        0.50 * 1.05;  // Robots cannot be closer than 50cm from the ball during a freekick
-                      // (with a buffer factor)
+        0.50 * 1.05;  // Robots cannot be closer than 50cm from the ball during a
+                      // free_kick (with a buffer factor)
```
Comment (jonathanlew) [inline src/software/ai/hl/stp/tactic/shadow_free_kicker_tactic.h]:
```suggestion
                      // free kick (with a buffer factor)
```

**Review by jonathanlew — CHANGES_REQUESTED** [review]
looks like you need to run fix formatting: https://github.com/UBC-Thunderbots/Software/blob/master/docs/workflow.md#formatting-your-code 

other than that, just a few nits

**Review by jonathanlew — APPROVED** [review]
LGTM!

**Review by kelvinkoon — APPROVED** [review]

**Pull request merged**